components/pkg-export-docker/src/build.rs

```diff
@@ -370,7 +373,7 @@ impl<'a> BuildSpec<'a> {
         let package_install = common::command::package::install::start(
             ui,
             url,
-            Some(channel),
+            Some(&ChannelIdent::from(channel)),
```

Comment:
Yeah, I was trying to figure out which places to change were a good return on investment and not waste too much time where the type wasn't valuable. I think I initially held off here because the internal types had `&str` fields and they can't be converted to `&ChannelIdents` because it requires an owned `String`. We could get fancy and change `ChannelIdent` to use `Cow` internally, but I think that's making things more complicated than necessary, so [I've just changed the `&str` fields to owned `ChannelIdent`s](https://github.com/habitat-sh/habitat/pull/6074/commits/4c12c3c93a0fc8a63cfb85ae966627ef65b0afdf).